### ISSUE TITLE
Consider using FMI4cpp

### ISSUE THREAD

**Issue by traversaro** [robotology/gazebo-fmi#27]
FMI4cpp is a new FMI 2.0 C++ library available at https://github.com/SFI-Mechatronics/FMI4cpp .

The API seems quite C++ idiomatic, and may be easier to use than the C-based api of [FMILibrary](https://jmodelica.org/fmil/FMILibrary-2.0.3-htmldoc/index.html) that we currently use.

**Comment by traversaro** [comment]
An interesting feature not available in FMILibrary would be to be able to call Model-Exchange models as Co-Simulation models  https://github.com/SFI-Mechatronics/FMI4cpp/issues/5 , this would give us the freedom of choose which kind of simulator use in the FMU exported from OpenModelica, that at the moment is constrained to be Forward Euler (see https://trac.openmodelica.org/OpenModelica/ticket/4273).
cc @triccyx @prashanthr05

**Comment by prashanthr05** [comment]
@traversaro thanks! looks like I need to catch up with this repository and its concepts ;D

**Comment by traversaro** [comment]
Yes! One of the advantages of https://github.com/SFI-Mechatronics/FMI4cpp is that is quite a new and self-contained codebase.

**Comment by traversaro** [comment]
Example of FMILibrary --> FMI4cpp migration: https://github.com/SFI-Mechatronics/FMU-proxy/pull/62 .
